src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -386,21 +405,34 @@ export default {
 			}
 		},
 
+		token() {
+			return this.$store.getters.getToken()
+		},
+
+		displayedMessage() {
+			switch (this.isFiltered) {
+			case 'mentions':
+				return t('spreed', 'You have no unread mentions in your inbox.')
+			case 'unread':
+				return t('spreed', 'You have no unread messages in your inbox.')
+			default:
+				return t('spreed', 'Your inbox is empty.')
+			}
+		},
+
 		filteredConversationsList() {
 			if (this.isFocused) {
 				return this.conversationsList
 			}
-
-			if (this.isFiltered === 'unread') {
-				return this.conversationsList.filter(conversation => conversation.unreadMessages > 0
-				|| conversation.hasCall || conversation.token === this.$store.getters.getToken())
-			}
-
-			if (this.isFiltered === 'mentions') {
-				return this.conversationsList.filter(conversation => conversation.unreadMention
-				|| conversation.hasCall
-				|| (conversation.unreadMessages > 0 && (conversation.type === CONVERSATION.TYPE.ONE_TO_ONE || conversation.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER))
-				|| conversation.token === this.$store.getters.getToken())
+			// applying filters
+			if (this.isFiltered) {
+				const filteredConversations = this.conversationsList.filter((conversation) => filterFunction(this.isFiltered, conversation)
+					|| conversation.hasCall
+					|| conversation.token === this.token)
```

Comment:
```suggestion
let validConversationsCount = 0
				const filteredConversations = this.conversationsList.filter((conversation) => {
					const conversationIsValid = filterFunction(this.isFiltered, conversation)
				if (conversationIsValid) {
					validConversationsCount++
				}
					return conversationIsValid
						|| conversation.hasCall
						|| conversation.token === this.token)
					}
```

Thus we don't need to filter the whole list again to get it's length (L434)

Reply:
Nice workaround 👍🏻